pkgs/os-specific/darwin/apple-sdk-11.0/cf-setup-hook.sh

```diff
@@ -1,6 +1,14 @@
 forceLinkCoreFoundationFramework() {
-  NIX_CFLAGS_COMPILE="-F@out@/Library/Frameworks${NIX_CFLAGS_COMPILE:+ }${NIX_CFLAGS_COMPILE-}"
-  NIX_LDFLAGS+=" @out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
+  local role_post
+  getHostRole
+
+  local cflags_compile_role_var="NIX_CFLAGS_COMPILE${role_post}"
+  local cflags_compile_role="${!cflags_compile_role_var-}"
+  export NIX_CFLAGS_COMPILE${role_post}="-F@out@/Library/Frameworks${cflags_compile_role:+ $cflags_compile_role}"
+
+  local ldflags_role_var="NIX_LDFLAGS${role_post}"
+  local ldflags_role="${!ldflags_role_var-}"
+  export NIX_LDFLAGS${role_post}+="${ldflags_role:+ }@out@/Library/Frameworks/CoreFoundation.framework/CoreFoundation.tbd"
 }
 
-preConfigureHooks+=(forceLinkCoreFoundationFramework)
```

Comment:
What's the motivation for this change? Not saying it's wrong, just not understanding it.

Reply:
`preConfigureHooks` move the execution of the function in another scope. `getHostRole` and as a result `role_post` are only available in this scope.

Reply:
When does this run when not explicitly added to one of the *Hooks variables?

Reply:
Actually, my description was slightly incorrect. Here's where it's executed:

https://github.com/NixOS/nixpkgs/blob/fec2bb16200119774e6f34e5d1d65d3a5eb7c61f/pkgs/stdenv/generic/setup.sh#L669-L671

`getHostRole` requires `$hostOffset` which is not available in the configuration phase. It needs to run in the scope of `activatePackage`.

Reply:
`activatePackage` runs for every build input in your derivation. Its purpose is to configure `PATH` and run setup hooks in the context of your derivation. This means the setup hook is aware of how its being included in the derivation (either through `buildInputs` or `nativeBuildInput`). That context is lost when you postpone the execution of the hook to `preConfigure`.

I'm not sure why this was originally configured in the `preConfigureHooks`. But I think configuring compiler/linker flags here should be fine. That's what setup hooks are for, no?